applications/GeoMechanicsApplication/custom_elements/geo_linear_truss_element.cpp

```diff
@@ -0,0 +1,220 @@
+#include "geo_mechanics_application_variables.h"
+#include "custom_utilities/structural_mechanics_element_utilities.h"
+#include "includes/checks.h"
+
+namespace Kratos {
+
+//----------------------------------------------------------------------------------------
+template< unsigned int TDim, unsigned int TNumNodes >
+GeoLinearTrussElement<TDim,TNumNodes>::
+    GeoLinearTrussElement(IndexType NewId,
+                          GeometryType::Pointer pGeometry)
+    : GeoTrussElementLinearBase<TDim,TNumNodes>(NewId, pGeometry) {}
+
+//----------------------------------------------------------------------------------------
+template< unsigned int TDim, unsigned int TNumNodes >
+GeoLinearTrussElement<TDim,TNumNodes>::
+    GeoLinearTrussElement(IndexType NewId,
+                          GeometryType::Pointer pGeometry,
+                          PropertiesType::Pointer pProperties)
+    : GeoTrussElementLinearBase<TDim,TNumNodes>(NewId, pGeometry, pProperties) {}
+
+//----------------------------------------------------------------------------------------
+template< unsigned int TDim, unsigned int TNumNodes >
+Element::Pointer GeoLinearTrussElement<TDim,TNumNodes>::
+    Create(IndexType NewId,
+           NodesArrayType const& rThisNodes,
+           PropertiesType::Pointer pProperties) const
+{
+    const GeometryType& rGeom = this->GetGeometry();
+    return Kratos::make_intrusive<GeoLinearTrussElement>(NewId, rGeom.Create(rThisNodes), pProperties);
+}
+
+//----------------------------------------------------------------------------------------
+template< unsigned int TDim, unsigned int TNumNodes >
+Element::Pointer GeoLinearTrussElement<TDim,TNumNodes>::
+    Create(IndexType NewId,
+            GeometryType::Pointer pGeom,
+            PropertiesType::Pointer pProperties) const
+{
+    return Kratos::make_intrusive<GeoLinearTrussElement>(NewId, pGeom, pProperties);
+}
+
+//----------------------------------------------------------------------------------------
+template< unsigned int TDim, unsigned int TNumNodes >
+GeoLinearTrussElement<TDim,TNumNodes>::~GeoLinearTrussElement() {}
+
+//----------------------------------------------------------------------------------------
+template< unsigned int TDim, unsigned int TNumNodes >
+void GeoLinearTrussElement<TDim,TNumNodes>::
+    ResetConstitutiveLaw()
+{
+    KRATOS_TRY
+
+    mInternalStresses = ZeroVector(mStressVectorSize);
+    mInternalStressesFinalized = ZeroVector(mStressVectorSize);
+    mInternalStressesFinalizedPrevious = ZeroVector(mStressVectorSize);
+
+    KRATOS_CATCH( "" )
+}
+
+//----------------------------------------------------------------------------------------
+template< unsigned int TDim, unsigned int TNumNodes >
+void GeoLinearTrussElement<TDim,TNumNodes>::
+    Initialize(const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+    GeoTrussElementLinearBase<TDim,TNumNodes>::Initialize(rCurrentProcessInfo);
+
+    mIsInitialization = true;
```

Comment:
why the extra flag/effort?
cannot you do here what you need to do?

Reply:
@philbucher Thanks for the suggestion! removed and it looks much nicer now!